usefulUtilities/loudApp/appResources/loudApp_app.js

```diff
@@ -0,0 +1,559 @@
+        }
+    };
+    var DEBUG = true;
+
+    // #region AUDIO
+
+    // Updates avatar's audio level in the userStore
+    var AVERAGING_RATIO = 0.05,
+        LOUDNESS_FLOOR = 11.0,
+        LOUDNESS_SCALE = 2.8 / 5.0,
+        LOG2 = Math.log(2.0),
+        AUDIO_PEAK_DECAY = 0.02;
+    function updateAudioForAvatar(uuid) {
+        if (!userStore[uuid]) {
+            return;
+        }
+        // scale audio
+        function scaleAudio(val) {
+            var audioLevel = 0.0;
+            if (val <= LOUDNESS_FLOOR) {
+                audioLevel = val / LOUDNESS_FLOOR * LOUDNESS_SCALE;
+            } else {
+                audioLevel = (val - (LOUDNESS_FLOOR - 1)) * LOUDNESS_SCALE;
+            }
+            if (audioLevel > 1.0) {
+                audioLevel = 1;
+            }
+            return audioLevel;
+        }
+        // the VU meter should work similarly to the one in AvatarInputs: log scale, exponentially averaged
+        // But of course it gets the data at a different rate, so we tweak the averaging ratio and frequency
+        // of updating (the latter for efficiency too).
+        var audioLevel = 0.0;
+        var avgAudioLevel = 0.0;
+        if (userStore[uuid]) {
+            // we will do exponential moving average by taking some the last loudness and averaging
+            userStore[uuid].accumulatedLevel = AVERAGING_RATIO * (userStore[uuid].accumulatedLevel || 0) + (1 - AVERAGING_RATIO) * (userStore[uuid].audioLoudness);
+
+            // add 1 to insure we don't go log() and hit -infinity.  Math.log is
+            // natural log, so to get log base 2, just divide by ln(2).
+            audioLevel = scaleAudio(Math.log(userStore[uuid].accumulatedLevel + 1) / LOG2);
+
+            // decay avgAudioLevel
+            avgAudioLevel = Math.max((1 - AUDIO_PEAK_DECAY) * (userStore[uuid].avgAudioLevel || 0), audioLevel).toFixed(3);
+        }
+        userStore[uuid].audioLevel = audioLevel;
+        userStore[uuid].avgAudioLevel = avgAudioLevel;
+    }
+
+
+    // Solos target avatar 
+    // Mutes the old target avatar
+    var activeTargetUUID = null;
+    function startListeningToAvatar(targetUUID) {
+        if (DEBUG) {
+            console.log("startListeningToAvatar isListening" + settings.ui.isListening + "targetUUID" + targetUUID + "activeTargetUUID" + activeTargetUUID);
+        }
+        if (settings.ui.isListening) {
+            // is already listening to activeTargetUUID
+            // mute old target
+            userStore[activeTargetUUID].isToggled = false;
+            Audio.removeFromSoloList([activeTargetUUID]);
+        } else {
+            // turn on settings.ui.isListening
+            settings.ui.isListening = true;
+        }
+        // add new target
+        Audio.addToSoloList([targetUUID]);
+        userStore[targetUUID].isToggled = true;
+        activeTargetUUID = targetUUID;
+    }
+
+
+    // Stops soloing target avatar
+    function stopListening() {
```

Comment:
don't you need to call `stopListening()` when the user switches domains?

Reply:
Already happens, just tested!